src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -965,18 +979,44 @@ def getParentDatasetTrio(self, childDataset):
           - file ids, run number and lumi section
         NOTE: This API is meant to be used by the StepChainParentage thread only!!!
         :param childDataset: name of the child dataset
-        :return: a dictionary where the key is a set of run/lumi, its value is the fileid
+        :return: A dictionary of named tuples of type FileRunLumi, using (run, lumi) tuples as keys
+                 {(1, 5110): FileRunLumi(fileId=2746490237, run=1, lumi=5110),
+                  (1, 5959): FileRunLumi(fileId=2746487877, run=1, lumi=5959),
+                  (1, 5990): FileRunLumi(fileId=2746487877, run=1, lumi=5990),
+                  ...}
         """
-        # this will return data in the format of:
+        # the call to DBS from bellow will return data in the following format of:
         # {554307997: [[1, 557179], [1, 557178],...
         # such that: key is file id, in each list is [run_number, lumi_section_numer].
         parentFullInfo = self.dbs.listParentDSTrio(dataset=childDataset)
 
-        # runs the actual mapping logic, like {"child_id": ["parent_id", "parent_id2", ...], etc
-        parentFrozenData = {}
-        for item in parentFullInfo:
-            for fileId in item:
-                for runLumiPair in item[fileId]:
-                    frozenKey = frozenset(runLumiPair)
-                    parentFrozenData[frozenKey] = fileId
-        return parentFrozenData
+        parentFlatData = {}
+        for fileId in parentFullInfo[0]:
+            for runLumiPair in parentFullInfo[0][fileId]:
+                frlObj = FileRunLumi(fileId=fileId, run=runLumiPair[0], lumi=runLumiPair[1])
+                parentFlatData[(frlObj.run, frlObj.lumi)] = frlObj
+        return parentFlatData
+
+    def getChildBlockTrio(self, childBlock):
+        """
+        Provided a dataset name, return all the parent dataset information, such as:
```

Comment:
The comment says dataset while function accepts block name

Reply:
fixed